### ISSUE TITLE
[Bug] Flawed TextLayer strings when getText is updated

### ISSUE THREAD

**Issue by DanSchoppe** [visgl/deck.gl#7768]
### Description

👋 Hello! Thank you for all your work on this library!

I bumped into an issue with `TextLayer`. If I instantiate a `TextLayer`, then update the `getText` accessor function such that **the fetched text changes in length**, the rendering becomes buggy.

Here's a demo:
https://user-images.githubusercontent.com/7377524/225937779-779b04d4-4b70-4790-9b48-75d22c78d13e.mp4

Observe how the `TextLayer` renders properly as long as, for each text object, the text fetched via `getText` doesn't change in length. However, once it encounters text with a different length than it rendered initially (in the demo, `"bagel"` is longer than the original `"ccc"` string), things start to behave strangely.

After looking through the source, I see that TextLayer keeps a `startIndices` array that's updated in `_updateText`. I'm guessing there's some issue with the imperative call to `_updateText` when `getText` changes.

### Flavors

- [X] React
- [ ] Python/Jupyter notebook
- [ ] MapboxLayer
- [ ] GoogleMapsOverlay
- [ ] CartoLayer
- [ ] DeckLayer/DeckRenderer for ArcGIS

### Expected Behavior

Strings rendered via `TextLayer` can vary in length over the lifecycle of a `TextLayer` instance.

### Steps to Reproduce

Codesandbox reproduction:
https://codesandbox.io/s/deck-gl-textlayer-gettext-length-bug-simplified-to2ert

### Environment

- Framework version: 8.9.2
- Browser: Chrome
- OS: MacOS

**Comment by Pessimistress** [comment]
I think the problem is here:

https://github.com/visgl/deck.gl/blob/d3a590c15b53c4b140ece8c08185cfe109ee3910/modules/layers/src/text-layer/text-layer.ts#L609-L616

All of these accessors should be invalidated if text length is changed, not just `getIcon`.
It should be something like:

```js
updateTriggers: {
  all: updateTriggers.getText,
  ...
}
```

It would be great if you can give it a test and open a PR!

**Comment by DanSchoppe** [comment]
Thanks for the tip @Pessimistress!

Just like you mentioned in #7769, I struggled to get `yarn link` working due to mismatched dependency versions. I'm working to recreating the bug via `yarn start-local` layer examples, then I'll investigate the `updateTriggers` you pointed out. If all goes well I'll work out a test and PR a fix.

Stay tuned!

**Comment by DanSchoppe** [comment]
This is enough to fix the problematic behavior I've experienced:
```diff
           updateTriggers: {
             getIcon: updateTriggers.getText,
-            getPosition: updateTriggers.getPosition,
+            getPosition: [updateTriggers.getPosition, updateTriggers.getText],
```

... but if you're sure the remaining props need to be invalidated as well (`getAngle`, `getColor`, etc etc) I can go with your `all: updateTriggers.getText` suggestion.

I'll also start working on a test. @Pessimistress do you mind pointing me to a file you'd imagine this test would belong in? Thanks!

**Comment by Pessimistress** [comment]
I'm pretty sure that the other props (color etc.) are not showing problems because you are using a constant. If you change `getColor` to a function you should observe the same bug.

The tests for this module are located in `test/modules/layers`. You can do something similar to https://github.com/visgl/deck.gl/blob/master/test/modules/layers/text-layer/text-layer.spec.js#L79 where you can create a layer and update its props over multiple test cases. After each update you can inspect the attributes like so:

```
      onAfterUpdate: ({layer, subLayer}) => {
        const positions = subLayer.getAttributeManager().instancePositions;
        // inspect positions.value
      }
```

**Comment by DanSchoppe** [comment]
@Pessimistress I'm having trouble getting tests to pass on my local MacBook M1 Max machine (compared to [this CI run](https://github.com/visgl/deck.gl/actions/runs/4450287341/jobs/7815560216), they hang before `# standalone#DeckGL`, perhaps similar to this person's experience: https://github.com/visgl/deck.gl/issues/7352#issuecomment-1287860000). To keep this issue moving along, I was wondering if it's possible to:
- run only the specific tests in `test/modules/layers`
- lean on CI to run the remaining tests. Is it a bad idea to enable GitHub Actions workflows on my forked repo?
<img width="733" alt="image" src="https://user-images.githubusercontent.com/7377524/226375336-eb768955-bcb7-4ece-8eb4-865b53b00007.png">

**Comment by Pessimistress** [comment]
You can run a specific test with `test.only(...)` instead of `test(...)`.

**Comment by Pessimistress** [comment]
Fixed included in v8.9.4
